libparsec/crates/types/src/local_manifest.rs

```diff
@@ -821,13 +830,35 @@ impl LocalFolderManifest {
         self
     }
 
+    // TODO: FINISH DOC !!!
+
+    /// Restore in the current manifest all the entries that were confined (according to
+    /// the given `prevent_sync_pattern`) in the `other` manifest.
+    ///
+    /// Note `other` is expected to be up to date with `prevent_sync_pattern` (i.e.
+    /// `apply_prevent_sync_pattern()` should have been called on it with this pattern).
```

Comment:
No it's not expected to be up-to-date: it's even used in `apply_prevent_sync_pattern` to restore previously confined entries that are no longer confined. This is why a timestamp is required: new entries might get added

Reply:
:+1: 
I've updated the comment:
```rust
    /// - `other` doesn't need to be up to date with `prevent_sync_pattern` (i.e.
    ///   `apply_prevent_sync_pattern()` may have been called on it with a different
```

Reply:
On top of that, if `prevent_sync_pattern` can mismatch then I guess this sentence is wrong:
```rust
    /// Restore in the current manifest all the entries that were confined (according to
    /// the given `prevent_sync_pattern`) in the `other` manifest.
```

The current manifest gets filled with confined entries from `other` (whatever the prevent sync pattern used to generate the confined entries list in `other`).

And given `self.remote_confinement_points` is also used, I guess the comment can be turned into:
```rust
     /// Restore in the current manifest all the entries that:
    /// - Were confined in `other` (according to `other`'s `local_confinement_points`).
    /// - Were in `other` but are now part of our remote confinement points.
    ///
    /// On top of that this method also apply `prevent_sync_pattern` on the restored
    /// entries.
    ///
    /// This method can be seen as the opposite of `filter_local_confinement_points`.
    ///
    /// Notes:
    /// - This method must be called only after `filter_remote_entries`, otherwise
    ///   our `remote_confinement_points` field may be out-of-sync with
    ///   `prevent_sync_pattern` and lead to inconsistent results.
    /// - `other` doesn't need to be up to date with `prevent_sync_pattern` (i.e.
    ///   `apply_prevent_sync_pattern()` may have been called on it with a different
    ///   prevent sync pattern).
    /// - `timestamp` is only used to update the `updated` field if a previously
    ///   confined entry is now not confined (and hence should now be synchronized !).
    ///
    ///
    /// For example, considering `.tmp` as prevent sync pattern and a manifest with:
    /// - `remote_confinement_points`: [1, 2]
    /// - `children`: `{"3.txt": 3}`
    /// - `base.children`: `{"1.tmp": 1, "2.tmp": 2, "3.txt": 3}`
    ///
    /// And an `other` manifest with:
    /// - `local_confinement_points`: [4]
    /// - `children`: `{"3.txt": 3, "2-renamed.txt": 2, "a.tmp": 4, "b.txt": 5}`
    ///
    /// Then the resulting manifest would have:
    /// - `base.children`: `{"1.tmp": 1, "2.tmp": 2, "3.txt": 3}` (no changes)
    /// - `remote_confinement_points`: [1, 2] (no changes)
    /// - `local_confinement_points`: []
    /// - `children`: `{"3.txt": 3, "2-renamed.txt": 2, "a.tmp": 4}`
    ///
    /// Note that `"b.txt": 5` is not restored as it is a local change that has nothing
    /// to do with confinement. Hence the output of this method is not to be considered
    /// as a merge with `other` (for that see `libparsec_client`'s `merge_local_folder_manifest`).
```

Reply:
I think `a.tmp` would end up in the local confinement points:
```
    /// Then the resulting manifest would have:
    /// - `base.children`: `{"1.tmp": 1, "2.tmp": 2, "3.txt": 3}` (no changes)
    /// - `remote_confinement_points`: [1, 2] (no changes)
    /// - `local_confinement_points`: [4]
    /// - `children`: `{"3.txt": 3, "2-renamed.txt": 2, "a.tmp": 4}`
```